### ISSUE TITLE
`ggman relocate` error after `ggman link`

### ISSUE THREAD

**Issue by janezicmatej** [tkw1536/ggman#6]
Steps to recreate:

1. run `ggman link` on some repository, eg. `ggman link $HOME/.config/nvim` links into `$GGROOT/github.com/janezicmatej/init.lua`
2. run `ggman relocate` to get this error

```sh
mkdir -p /Users/janezicmatej/Desktop/git/github.com/janezicmatej
mv /Users/janezicmatej/.config/nvim /Users/janezicmatej/Desktop/git/github.com/janezicmatej/init.lua
Unable to move repository: Rename /Users/janezicmatej/.Config/nvim
/Users/janezicmatej/Desktop/git/github.Com/janezicmatej/init.Lua: Not a
directory
```

**Comment by tkw1536** [comment]
Sorry for the late reply, for some reason I overlooked the notification. 

I have never used `ggman relocate` in conjunction with `ggman link`.

I believe the problem is that when running `ggman relocate` all repositories to be moved are resolved absolutely first, and I did not consider the case when it is a symlinked repository in the first place.
I will see if I can introduce a special case for it.

**Comment by janezicmatej** [comment]
I've used ggman for about half a year now. I find it most useful for `ggcd`, `ggclone` and having a `$GGHOME/no-remote` directory where I make a new folders and `git init` in them. When I feel like something is not just another throw away project and I want to push it I just add a remote and `ggman relocate`. It really is a great tool!

**Comment by tkw1536** [comment]
I have pushed a first fix (38f177ef07f92ae02bce6e872cc11119c9d764ef) for the issue described.

I have fixed the case where there are symlinked repositories under `$GGROOT` that are already in the correct place (like those coming from`ggman link`). 

However I have not yet fixed the case where a linked repo is the wrong place and should be moved.
That requires keeping track of the original (non-`realpath`ed) path through the repository scanning logic, and will require a bit more thinking to get right.
